addons/barcodes/static/src/js/barcode_events.js

```diff
@@ -0,0 +1,148 @@
+odoo.define('web.BarcodeEvents', function(require) {
+"use strict";
+
+var core = require('web.core');
+var mixins = core.mixins;
+
+var BarcodeEvents = core.Class.extend(mixins.PropertiesMixin, {
+    timeout: null,
+    key_pressed: {},
+    buffered_key_events: [],
+    // Regexp to match a barcode input and extract its payload
+    // Note: to build in init() if prefix/suffix can be configured
+    regexp: /(.{3,})[\n\r\t]+/,
```

Comment:
What is the point of introducing the regexp? What problem does this solve?

Also, if this is necessary, then this regexp should be comprised of `(.{3,}) + suffix` to avoid having the suffix copy pasted in two different places. Check [this](http://stackoverflow.com/questions/185510/how-can-i-concatenate-regex-literals-in-javascript) for the best way to concatenate regular expression in js.